src/mongoc/mongoc-topology-scanner.c

```diff
@@ -167,7 +171,7 @@ mongoc_topology_scanner_rm (mongoc_topology_scanner_t *ts,
 
    ele = mongoc_topology_scanner_get_node (ts, id);
    if (ele) {
-      mongoc_topology_scanner_node_destroy (ele);
+      mongoc_topology_scanner_node_destroy (ele, true);
```

Comment:
It looks like scanner_rm never gets called anywhere.  I'd just get rid of the whole function